test/NuGet.Core.Tests/NuGet.Common.Test/ConcurrencyUtilitiesTests.cs

```diff
@@ -13,37 +15,57 @@ public class ConcurrencyUtilitiesTests
         public async Task ConcurrencyUtilities_LockStress()
         {
             // Arrange
-            var sem = new ManualResetEventSlim();
-            var threads = 1000;
-            var tasks = new Stack<Task<bool>>(threads);
-            var path = Path.Combine(Path.GetTempPath(), "ConcurrencyUtilities_LockStress");
-            var token = CancellationToken.None;
-            Func<CancellationToken, Task<bool>> action = (ct) => {
-                // Wait till all threads are ready
-                sem.Wait();
-                return Task.FromResult(true);
-            };
-
-            while (tasks.Count < threads)
-            {
-                var task = Task.Run(async () => await ConcurrencyUtilities.ExecuteWithFileLockedAsync<bool>(
-                    path,
-                    action,
-                    token));
-
-                tasks.Push(task);
-            }
-
-            // Act
-            // Release all the threads at once
-            sem.Set();
-            await Task.WhenAll(tasks);
-
-            // Assert
-            while (tasks.Count > 0)
+            using(var testDirectory = TestFileSystemUtility.CreateRandomTestFolder())
             {
-                // Verify everything finished without errors
-                Assert.True(await tasks.Pop());
+                // This is the path that uniquely identifies the system-wide mutex.
+                var path = Path.Combine(testDirectory, "ConcurrencyUtilities_LockStress_Verification");
+                
+                // This is a semaphore use to verify the lock.
+                var verificationSemaphore = new SemaphoreSlim(1);
+                
+                // Iterate a lot, to increase confidence.
+                const int threads = 50;
+                const int iterations = 10;
+                
+                // This is the action that is execute inside of the lock.
+                Func<CancellationToken, Task<bool>> lockedActionAsync = async lockedToken =>
+                {
+                    var acquired = await verificationSemaphore.WaitAsync(0);
+                    if (!acquired)
+                    {
+                        throw new Exception("The semaphore could not be acquired, meaning the locking mechanism is not working.");
```

Comment:
Sure, I can return `false` here, although it seems like throwing exceptions in tests can be useful sometimes.